internal/controller/testing.go

```diff
@@ -66,3 +67,13 @@ func (tc *TestController) Runtime() Runtime {
 		Cache:  tc.cache,
 	}
 }
+
+// DryRunMapper will trigger the appropriate DependencyMapper for an update of
+// the provided type and return the requested reconciles.
+//
+// Useful for testing just the DependencyMapper+Cache interactions for chains
+// that are more complicated than just a full controller interaction test would
+// be able to easily verify.
+func (tc *TestController) DryRunMapper(ctx context.Context, res *pbresource.Resource) ([]Request, error) {
```

Comment:
Adding this let me be more confident in the convoluted mess of relationships between the mappers and the cache.